model/project_ref.go

```diff
@@ -447,6 +447,11 @@ func (p *ProjectRef) Add(creator *user.DBUser) error {
 		}
 	}
 
+	// Default to main for branch
+	if p.Branch == "" {
```

Comment:
considering this change, should we add "branch" to the create project modal so we give users a chance to indicate a branch before defaulting to main?
![image](https://user-images.githubusercontent.com/13104717/182448536-914d5d1b-442f-4460-a045-5a2c2b6d3388.png)

Reply:
That sounds reasonable to me, I opened [EVG-17391](https://jira.mongodb.org/browse/EVG-17391).

I wrote optional in the ticket but if we go this route, the branch could be a required field that prevents submission if it's empty. We could also pre-fill it to say main, which would kind of duplicate the work in this ticket. I'm not sure which approach is better 🤔

Reply:
Either approach seems reasonable to me. I'll defer to Annie and Sophie.

Reply:
I would say no personally. I'd prefer to just remove the default than add it to the modal.

Reply:
Oh I see now that we _had_ to add a default because spruce needs it. This now seems like a design question to me about what would be least confusing to users. I personally think it's fine to just always default to main though; it's simpler, unclutters the create project modal, and if users know they need a certain branch then they'll change it.

Reply:
In that case, instead of defaulting, should we maybe just make the text box yell at users so they can see why they can't save the page, similar to what we do here?
![image](https://user-images.githubusercontent.com/13104717/182458475-fdd42f11-0b09-4e9f-b52c-2426750860f9.png)

Reply:
or that. I typed my response before seeing the above.